### PR TITLE
Remove Captcha challenge: none and use status() to en/disable Captchas

### PR DESCRIPTION
**Pull request by LKS90**
This PR makes the CaptchaPoint actually use the status attribute which it inherits from Entity. I removed the challenge type `none` and use the status to decide whether the captcha is active on a form.

I updated the tests so the Captchas are enabled when they are being tested.

### CODE REVIEW

File: captcha.module [hunk 1]
```diff
@@ -327,13 +328,17 @@ function captcha_form_alter(array &$form, FormStateInterface $form_state, $form_
...
 
-    if ($captcha_point !== NULL && $captcha_point->getCaptchaType()) {
+    if (!empty($captcha_point) && $captcha_point->status()) {
       $markup = '<a href="@change">change</a>';
       $captcha_element['#title'] = $captcha_point->status() ? t('CAPTCHA: challenge "@type" enabled', ['@type' => $captcha_point->getCaptchaType()]) : t('CAPTCHA: challenge "@type" disabled', ['@type' => $captcha_point->getCaptchaType()]);
-      $captcha_point->status() ? $captcha_element['#description'] = t('Untrusted users will see a CAPTCHA here (<a href="@settings">general CAPTCHA settings</a>).',
-        ['@settings' => Url::fromRoute('captcha_settings')->toString()]) : $captcha_element['#description'] = t('CAPTCHA disabled, Untrusted users won\'t see the captcha (<a href="@settings">general CAPTCHA settings</a>).',
-        ['@settings' => Url::fromRoute('captcha_settings')->toString()]
-        );
+      if ($captcha_point->status()) {
```
Comment (Berdir) [inline captcha.module]: You never get into the else case anymore because you call status() already on the outer if.

File: src/Tests/CaptchaBaseWebTestCase.php [hunk 2]
```diff
@@ -101,6 +101,7 @@ public function setUp() {
     $comment_field = FieldConfig::loadByName('node', 'page', 'comment');
     $comment_field->setSetting('form_location', CommentItemInterface::FORM_SEPARATE_PAGE);
     $comment_field->save();
+    $this->config('captcha.settings')->set('default_challenge', 'captcha/test')->save();
```
Comment (Berdir) [inline src/Tests/CaptchaBaseWebTestCase.php:108]: Why is this needed now?

File: src/Tests/CaptchaCacheTestCase.php [hunk 3]
```diff
@@ -42,7 +42,10 @@ public function testCacheTags() {
     $this->assertEqual($this->drupalGetHeader('x-drupal-cache'), 'HIT');
 
     // Enable captcha on login block and test caching.
-    captcha_set_form_id_setting('user_login_form', 'captcha/Math');
+    /* @var \Drupal\captcha\Entity\CaptchaPoint $captcha_point */
+    $captcha_point = \Drupal::entityManager()->getStorage('captcha_point')->load('user_login_form');
+    $captcha_point->setCaptchaType('captcha/Math');
+    $captcha_point->enable()->save();
```
Comment (Berdir) [inline src/Tests/CaptchaCacheTestCase.php]: What about making sure that helper function enables() captcha points as well? less changes for this PR.

**Comment by LKS90** [comment]
Can't reproduce the test fails locally, will have to have another look at this.

**Comment by LKS90** [comment]
The test fails were random, reverted a small change which isn't related to this task and the travis_ci build passes :).
